pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,101 @@
 package ossm
 
 import (
-	"strings"
+	_ "embed"
+	"fmt"
 	"testing"
-	"time"
 
-	"github.com/maistra/maistra-test-tool/pkg/util"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
 	"github.com/maistra/maistra-test-tool/pkg/util/env"
-	"github.com/maistra/maistra-test-tool/pkg/util/log"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
-func installDefaultSMCP23() {
-	log.Log.Info("Create SMCP v2.3 in ", meshNamespace)
-	util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-	util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV23_template, smcp))
-	util.KubeApplyContents(meshNamespace, smmr)
-
-	// patch SMCP identity if it's on a ROSA cluster
-	if env.Getenv("ROSA", "false") == "true" {
-		util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-	}
-	log.Log.Info("Waiting for mesh installation to complete")
-	util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-	time.Sleep(time.Duration(20) * time.Second)
+type vars struct {
+	Name      string
+	Namespace string
 }
 
-func installDefaultSMCP22() {
-	log.Log.Info("Create SMCP v2.2 in ", meshNamespace)
-	util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-	util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-	util.KubeApplyContents(meshNamespace, smmr)
-
-	// patch SMCP identity if it's on a ROSA cluster
-	if env.Getenv("ROSA", "false") == "true" {
-		util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
+func installSMCPVersion(t test.TestHelper, smcpTemplate string, vars interface{}) {
```

Comment:
I'd move this and the uninstall function below the test function, so that it's the first thing you see when you open the file (it's also the most important part of the file; and it shows at a high level what the test does... if someone then wants to drill down to see what `installSMCPVersion` or `uninstallSMCPVersion` functions do, they can scroll down (or click them wherever they get called).

Reply:
Done